File: backend/src/main/java/com/taskflow/task/dto/CreateTaskRequest.java
package com.taskflow.task.dto;

import com.taskflow.task.TaskPriority;
import jakarta.validation.constraints.NotBlank;

import java.time.LocalDate;
import java.util.UUID;

public class CreateTaskRequest {

    @NotBlank(message = "is required")
    private String title;
    private String description;
    private TaskPriority priority = TaskPriority.medium;
    private UUID assigneeId;
    private LocalDate dueDate;

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public TaskPriority getPriority() { return priority; }
    public void setPriority(TaskPriority priority) { this.priority = priority; }
    public UUID getAssigneeId() { return assigneeId; }
    public void setAssigneeId(UUID assigneeId) { this.assigneeId = assigneeId; }
    public LocalDate getDueDate() { return dueDate; }
    public void setDueDate(LocalDate dueDate) { this.dueDate = dueDate; }
}

File: backend/apps/tasks/requests/create_task_request.go
package requests

type CreateTaskRequest struct {
	Title       string  `json:"title"       validate:"required"`
	Description *string `json:"description"`
	Status      string  `json:"status"      validate:"omitempty,oneof=todo in_progress done"`
	Priority    string  `json:"priority"    validate:"omitempty,oneof=low medium high"`
	AssigneeID  *string `json:"assignee_id" validate:"omitempty,uuid"`
	DueDate     *string `json:"due_date"`
}
